pkg/kubelet/dockertools/runner.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockertools
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/capabilities"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+	kubecontainer "github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/container"
+	"github.com/fsouza/go-dockerclient"
+	"github.com/golang/glog"
+)
+
+type DockerContainerRunner struct {
+	Client   DockerInterface
+	Recorder record.EventRecorder
```

Comment:
@yifan-gu i moved that back to kubelet runContainer
On Fri, Mar 27, 2015 at 6:59 PM Yifan Gu <notifications@github.com> wrote:

> In pkg/kubelet/dockertools/runner.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6100#discussion_r27337081>
> :
>
> > +import (
> > +	"fmt"
> > +	"os"
> > +	"path"
> > +
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/capabilities"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
> > +	kubecontainer "github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/container"
> > +	"github.com/fsouza/go-dockerclient"
> > +	"github.com/golang/glog"
> > +)
> > +
> > +type DockerContainerRunner struct {
> > +	Client   DockerInterface
> > +	Recorder record.EventRecorder
>
> @pmorie <https://github.com/pmorie> Yes. But we missed the SetRef part
> here after container creation, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6100/files#r27337081>
> .
>